server/schedule/region_scatterer.go

```diff
@@ -33,59 +33,108 @@ const regionScatterName = "region-scatter"
 type selectedLeaderStores struct {
 	mu     sync.Mutex
 	stores map[uint64]uint64 // storeID -> hintCount
+	// TODO: support auto-gc for the groupDistribution
+	groupDistribution map[string]map[uint64]uint64 // group -> StoreID -> leaderCount
 }
 
-func (s *selectedLeaderStores) put(id uint64) {
+func (s *selectedLeaderStores) put(id uint64, group string) {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 	s.stores[id] = s.stores[id] + 1
+	if group != "" {
+		distribution, ok := s.groupDistribution[group]
+		if !ok {
+			distribution = make(map[uint64]uint64)
+		}
+		distribution[id] = distribution[id] + 1
+		s.groupDistribution[group] = distribution
+	}
 }
 
-func (s *selectedLeaderStores) get(id uint64) uint64 {
+func (s *selectedLeaderStores) get(id uint64, group string) uint64 {
 	s.mu.Lock()
 	defer s.mu.Unlock()
-	return s.stores[id]
+	if group == "" {
+		return s.stores[id]
+	}
+	distribution, ok := s.groupDistribution[group]
+	if !ok {
+		return 0
+	}
+	return distribution[id]
 }
 
 func newSelectedLeaderStores() *selectedLeaderStores {
 	return &selectedLeaderStores{
-		stores: make(map[uint64]uint64),
+		stores:            make(map[uint64]uint64),
+		groupDistribution: make(map[string]map[uint64]uint64),
 	}
 }
 
 type selectedStores struct {
-	mu     sync.Mutex
-	stores map[uint64]struct{}
+	mu sync.Mutex
+	// TODO: support auto-gc for the stores
+	stores map[string]map[uint64]struct{} // group -> StoreID -> struct{}
+	// TODO: support auto-gc for the groupDistribution
+	groupDistribution map[string]map[uint64]uint64 // group -> StoreID -> peerCount
 }
 
 func newSelectedStores() *selectedStores {
 	return &selectedStores{
-		stores: make(map[uint64]struct{}),
+		stores:            make(map[string]map[uint64]struct{}),
+		groupDistribution: make(map[string]map[uint64]uint64),
 	}
 }
 
-func (s *selectedStores) put(id uint64) bool {
+func (s *selectedStores) put(id uint64, group string) bool {
```

Comment:
Can merge `selectedStores` and `selectedLeaderStores `?

Reply:
updated.